src/services/item/plugins/etherpad/test/service-api.test.ts

```diff
@@ -0,0 +1,773 @@
+import { BOB, saveMember } from '../../../../member/test/fixtures/members';
+import { ItemRepository } from '../../../repository';
+import { MAX_SESSIONS_IN_COOKIE } from '../constants';
+import { EtherpadItemService } from '../service';
+import { setUpApi } from './api';
+
+// mock datasource
+jest.mock('../../../../../plugins/datasource');
+
+const MOCK_GROUP_ID = 'g.s8oes9dhwrvt0zif';
+const MOCK_PAD_READ_ONLY_ID = 'r.s8oes9dhwrvt0zif';
+const MOCK_PAD_NAME = 'mock-pad-name';
+const MOCK_PAD_ID = `${MOCK_GROUP_ID}\$${MOCK_PAD_NAME}`;
+const MOCK_AUTHOR_ID = 'a.s8oes9dhwrvt0zif';
+const MOCK_SESSION_ID = 's.s8oes9dhwrvt0zif';
+const MODES: Array<'read' | 'write'> = ['read', 'write'];
+
+describe('Service API', () => {
+  let app: FastifyInstance;
+  let member: Member;
+
+  const payloadCreate = {
+    method: HttpMethod.POST,
+    url: 'items/etherpad/create',
+    payload: {
+      name: 'test-item-name',
+    },
+  };
+
+  beforeEach(async () => {
+    let actor: Member | null;
+    ({ app, actor } = await build());
+    if (!actor) {
+      throw new Error('Test error: member should be defined');
+    }
+    member = actor;
+  });
+
+  afterEach(async () => {
+    jest.clearAllMocks();
+    await clearDatabase(app.db);
+    app.close();
+    nock.cleanAll();
+  });
+
+  describe('create a pad', () => {
+    it('creates a pad successfully', async () => {
+      const reqsParams = setUpApi({
+        createGroupIfNotExistsFor: [
+          StatusCodes.OK,
+          { code: 0, message: 'ok', data: { groupID: MOCK_GROUP_ID } },
+        ],
+        createGroupPad: [StatusCodes.OK, { code: 0, message: 'ok', data: null }],
+      });
+      const res = await app.inject(payloadCreate);
+
+      const { createGroupIfNotExistsFor, createGroupPad } = await reqsParams;
+      expect(createGroupPad?.get('groupID')).toEqual(MOCK_GROUP_ID);
+      // groupMapper sent to etherpad is equal to the generated padID
+      expect(createGroupIfNotExistsFor?.get('groupMapper')).toEqual(createGroupPad?.get('padName'));
+
+      expect(res.statusCode).toEqual(StatusCodes.OK);
+      expect(res.json()).toMatchObject({
+        name: 'test-item-name',
+        extra: {
+          etherpad: {
+            padID: `${MOCK_GROUP_ID}$${createGroupPad?.get('padName')}`,
+            groupID: MOCK_GROUP_ID,
+          },
+        },
+      });
+    });
+
+    it('returns error on etherpad HTTP error', async () => {
+      setUpApi({
+        createGroupIfNotExistsFor: [
+          StatusCodes.OK,
+          { code: 0, message: 'ok', data: { groupID: MOCK_GROUP_ID } },
+        ],
+        createGroupPad: [StatusCodes.GATEWAY_TIMEOUT],
+      });
+      const res = await app.inject(payloadCreate);
+
+      expect(res.statusCode).toEqual(StatusCodes.INTERNAL_SERVER_ERROR);
+      expect(res.json()).toMatchObject({
+        code: 'GPEPERR001',
+        message: 'Internal Etherpad server error',
+        origin: 'graasp-plugin-etherpad',
```

Comment:
Use constant for plugin name?

Reply:
In tests I prefer to use static literals to avoid any dynamic logic that may interfere with the test in the future